alf/networks/network.py

```diff
@@ -133,12 +226,49 @@ def copy(self, **kwargs):
     def __call__(self, inputs, *args, **kwargs):
         return super(Network, self).__call__(inputs, *args, **kwargs)
 
+    @property
+    def nested_input_tensor_spec(self):
+        """Return the input tensor spec BEFORE preprocessings have been applied.
+        """
+        return self._nested_input_tensor_spec
+
+    @property
+    def input_tensor_spec(self):
```

Comment:
I suggest this to be the original input tensor spec. i.e. (self._nested_input_tensor_spec) to reduce confusion. And it seems there is no need to expose the tensor spec after preprocessing.

Reply:
But then `self.input_tensor_spec` and `self._input_tensor_spec` return two different things. And the latter is always needed for constructing downstream networks as their input specs.

Reply:
Perhaps change to a different name so not to be confused? For example, processed_tensor_spec